### PR TITLE
Fix: Better play error handling and logging

### PR DESCRIPTION
**Pull request by krak3n**
This PR tweaks the logic around handling errors with playing tracks. Currently I have no idea what happens when a track fails to play so it will now log all exceptions so I can see whats going on and wrote better code around handling those exceptions.

Basically we will always block when we try and play a track, on the event of an exception we will follow the normal stop procedure which unloads the track and unblocks the watcher and should then play the next track and log the exception that occurred.

### CODE REVIEW

**Comment by gregatsoon** [comment]
:+1:

File: fmplayer/player.py [hunk 1]
```diff
@@ -121,24 +117,32 @@ def play(self, uri):
...
+        """ Fired when a playing track finishes, ensures the tack is unloaded
+        and the ``STOP_EVENT`` is set to ``True``.
         """
 
-        logger.debug('Track Stop - Unload')
-        self.session.player.unload()
-        logger.debug('STOP_EVENT set')
-        STOP_EVENT.set()  # Unblocks the playlist watcher
+        if session is None:
+            session = self.session
```
Comment (radeklos) [inline fmplayer/player.py:140]: What is difference between session and self.session? Spotify uses only one user instance so session should be same.

Comment (krak3n) [inline fmplayer/player.py:140]: Yup you are right, the stop the event gets the session passed into it in the event so I'm just preferring to use that when stop is called by the stop event. But yeah there should be no difference.

**Comment by krak3n** [comment]
Ooops closed this before review the comments :disappointed:

**Pull request merged**